packages/api-plugin-orders/src/util/addShipmentMethodToGroup.js

```diff
@@ -42,16 +42,18 @@ export default async function addShipmentMethodToGroup(context, {
   const rates = await queries.getFulfillmentMethodsWithQuotes(commonOrder, context);
   const errorResult = rates.find((option) => option.requestStatus === "error");
   if (errorResult) {
-    throw new ReactionError("invalid", errorResult.message);
+    const eventData = { field: "FulfillmentMethod", value: "Returned error" };
+    throw new ReactionError("invalid", errorResult.message, eventData);
   }
 
   const selectedFulfillmentMethod = rates.find((rate) => selectedFulfillmentMethodId === rate.method._id);
   if (!selectedFulfillmentMethod) {
+    const eventData = { field: "selectedFulfillmentMethodId", value: selectedFulfillmentMethodId };
     throw new ReactionError("invalid", "The selected fulfillment method is no longer available." +
-      " Fetch updated fulfillment options and try creating the order again with a valid method.");
+      " Fetch updated fulfillment options and try creating the order again with a valid method.", eventData);
   }
 
-  group.shipmentMethod = {
+  const output = {
```

Comment:
If this will no longer be mutating the group object, maybe it should be in a new function with a different name, such as `getShipmentMethodForGroup`?